### PR TITLE
Add support for custom `helpful-buffer-name-function's

### PR DESCRIPTION
**Pull request by whxvd**

### CODE REVIEW

**Comment by whxvd** [comment]
There are two rationales for the PR:

First, the default setting for e.g. `helm-buffers-list` truncates buffer names after 20 characters. A buffer list in that case may look as follows:
```
*helpful function: o...      954      Helpful  (in `~/scratch/')
*helpful function: o...     1.2k      Helpful  (in `~/scratch/')
```
This can be alleviated by modifying `helm-buffer-max-length`. But on an 80 character wide display with other columns shown for each buffer, the space available for a buffer names still is limited.

Second, the `helpful ` in the buffer name is superfluous in all of my use cases. When a buffer has a name like `*function: elispy-function-name*` that is enough to identify it as a Helpful buffer.

In short, the `helpful ` in Helpful buffer names is not necessarily helpful. :-)
